pinot-spi/src/main/java/org/apache/pinot/spi/data/FieldSpec.java

```diff
@@ -99,6 +100,7 @@ public abstract class FieldSpec implements Comparable<FieldSpec>, Serializable {
   protected String _name;
   protected DataType _dataType;
   protected boolean _isSingleValueField = true;
+  protected boolean _nullable = true;
```

Comment:
Since we use `notNull` as the key, can we also store `_notNull` internally and default to `false`? It is quite confusing to see it defined as `_nullable`, default to `true`, and has `isNullable()` and `setNullable()` json ignored

Reply:
We can store that attribute. But adding too many negations makes things difficult to understand. It is easier to read
```
if (field.isNullable()) {
```

Than

```
if (!field.isNotNull()) {
```

Reply:
Changed in 802e9c4f93caa0815e74d85035847c1a2153f20f